File: flink-basic/src/main/scala/com/xk/bigdata/flink/basic/scala/BatchWcApp.scala
package com.xk.bigdata.flink.basic.scala

import org.apache.flink.api.scala.ExecutionEnvironment
import org.apache.flink.api.scala._

/**
 * Scala Flink 批处理 WC
 */
object BatchWcApp {

  def main(args: Array[String]): Unit = {

    val env = ExecutionEnvironment.getExecutionEnvironment
    env.readTextFile("data/wc.txt")
      .flatMap(_.toLowerCase.split(","))
      .map((_, 1))
      .groupBy(0)
      .sum(1)
      .print()
  }

}

File: flink-basic/src/main/java/com/xk/bigdata/flink/basic/java/BatchWcApp.java
package com.xk.bigdata.flink.basic.java;

import org.apache.flink.api.common.functions.FlatMapFunction;
import org.apache.flink.api.common.functions.MapFunction;
import org.apache.flink.api.java.ExecutionEnvironment;
import org.apache.flink.api.java.tuple.Tuple2;
import org.apache.flink.util.Collector;

/**
 * Java Flink 批处理 WC
 */
public class BatchWcApp {

    public static void main(String[] args) throws Exception {

        ExecutionEnvironment env = ExecutionEnvironment.getExecutionEnvironment();
        env.readTextFile("data/wc.txt")
                .flatMap(new FlatMapFunction<String, String>() {
                    @Override
                    public void flatMap(String s, Collector<String> collector) throws Exception {
                        String[] words = s.toLowerCase().split(",");
                        for (String word : words) {
                            collector.collect(word);
                        }
                    }
                })
                .map(new MapFunction<String, Tuple2<String, Integer>>() {
                    @Override
                    public Tuple2<String, Integer> map(String s) throws Exception {
                        return new Tuple2<>(s, 1);
                    }
                })
                .groupBy(0)
                .sum(1)
                .print();
    }

}
